src/Kanbas/Courses/Quizzes/Preview/index.tsx

```diff
@@ -5,26 +5,94 @@ import QuizQuestionPreviewSingle from "./Question/Single";
 import QuizQuestionPreviewList from "./Question/List";
 import MiddleContentActions from "../../../layout/Content/NotLeftSide/MiddleContent/MiddleContentActions";
 import MiddleContentData from "../../../layout/Content/NotLeftSide/MiddleContent/MiddleContentData";
+import { IoAlertCircleOutline } from "react-icons/io5";
+import { useDispatch, useSelector } from "react-redux";
+import { KanbasState } from "../../../store";
+import { Button, Form } from "react-bootstrap";
+import "./index.css";
+import { useEffect, useRef, useState } from "react";
+import { IKanbasQuizQuestion } from "../../../store/interfaces/quizzes";
+import { useParams } from "react-router";
+import { getQuizQuestions } from "../client";
+import { resetPreview, setQuizQuestions } from "../reducer";
+import { getCurrentHumanReadableDate } from "../../common/Utils";
 
 function QuizPreview() {
 
-    // TODO this variable will be removed later to use props directly from JSON
-    const isQneQuestionAtATime = false;
+    const { quizId } = useParams();
+    const dispatch = useDispatch();
+    const questionRefs = useRef<(HTMLElement | null)[]>([]);
```

Comment:
We dont need a ref of the HTML Element here. Instead, we can just pass around the state of questions, or using redux, rely on the specific question that is passed to the child component. I'd recommend using a preview specific redux state as this is not needed in the other areas of the app.